### PR TITLE
Add spec for Ag (Silver Searcher)

### LINKED ISSUE #529: [ag] support for the_silver_searcher
### Sanity checks

- [X] My issue relates to a specific CLI completion spec (e.g. `git checkout` is missing options in `git` completion spec). If your issue is more general, please create your issue here: [withfig/fig](https://github.com/withfig/fig/issues/new?assignees=&labels=NEED_TO_LABEL&template=main_issue_template.yml)
- [X] I have searched [github.com/withfig/autocomplete/issues](https://github.com/withfig/autocomplete/issues?q=) and there are no duplicates of my issue
- [ ] I would like to work on this.

### What CLI tool does this relate to?

ag

### Which statement makes the most sense?

This completion spec does not exist

### Issue Details

ag (the_silver_searcher) is a faster version of ack

https://github.com/ggreer/the_silver_searcher

### PR DESCRIPTION
**Pull request by RSO**
**What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)**
Add all the args from the `man ag` page.

This is my first time writing autocomplete specs so let me know if there's anything I should've done differently!

Fixes #529

### CODE REVIEW

File: dev/ag.ts [hunk 1]
```diff
@@ -0,0 +1,433 @@
...
+    },
+
+    {
+      name: "--color",
+      description: "Print color codes in results.",
+    },
+
+    {
+      name: "--nocolor",
+      description: "Don't print color codes in results.",
```
Comment (QuiiBz) [inline dev/ag.ts:85]: We can add [exclusiveOn](https://fig.io/docs/reference/option#exclusiveon) for those two options!

File: dev/ag.ts [hunk 2]
```diff
@@ -0,0 +1,433 @@
...
+    },
+
+    {
+      name: ["-f", "--follow"],
+      description: "Follow symlinks.",
+    },
+
+    {
+      name: "--nofollow",
+      description: "Don't follow symlinks.",
```
Comment (QuiiBz) [inline dev/ag.ts:150]: Same as above for `filename / nofilename` and `follow / nofollow`

File: dev/ag.ts [hunk 3]
```diff
@@ -0,0 +1,433 @@
...
+    {
+      name: "--group",
+      description:
+        "Lumps multiple matches in the same file together, and presents them under a single occurrence of the filename.",
+    },
+
+    {
+      name: "--nogroup",
+      description:
+        "Refrains from lumping matches in the same file together, and instead places the filename at the start of each match line.",
```
Comment (QuiiBz) [inline dev/ag.ts:167]: Same as above

File: dev/ag.ts [hunk 4]
```diff
@@ -0,0 +1,433 @@
...
+    },
+
+    {
+      name: ["-H", "--heading"],
+      description: "Print filenames above matching contents.",
+    },
+
+    {
+      name: "--noheading",
+      description: "Don't print filenames above matching contents.",
```
Comment (QuiiBz) [inline dev/ag.ts:195]: Same as above

File: dev/ag.ts [hunk 5]
```diff
@@ -0,0 +1,433 @@
...
+      description: "Search hidden files. This option obeys ignored files.",
+    },
+
+    {
+      name: "--ignore",
+      description:
+        "Ignore files/directories whose names match this pattern. Literal file and directory names are also allowed.",
+      args: {
+        name: "PATTERN",
+        description: "The pattern to look for",
```
Comment (QuiiBz) [inline dev/ag.ts:209]: I think we can also add a template of `filepaths` and `folders` here

File: dev/ag.ts [hunk 6]
```diff
@@ -0,0 +1,433 @@
...
+      },
+    },
+
+    {
+      name: "--ignore-dir",
+      description: "Alias for --ignore for compatibility with ack.",
+      args: {
+        name: "NAME",
+        description: "The directory to ignore",
```
Comment (QuiiBz) [inline dev/ag.ts:218]: Same as above but only for `folers`

File: dev/ag.ts [hunk 7]
```diff
@@ -0,0 +1,433 @@
...
+    },
+
+    {
+      name: "--multiline",
+      description: "Match regexes across newlines.",
+    },
+
+    {
+      name: "--nomultiline",
+      description: "Don't match regexes across newlines",
```
Comment (QuiiBz) [inline dev/ag.ts:267]: Same as above for [exclusiveOn](https://fig.io/docs/reference/option#exclusiveon)

File: dev/ag.ts [hunk 8]
```diff
@@ -0,0 +1,433 @@
...
+      description:
+        "When  recursing  directories, don't scan dirs that reside on other storage devices. This lets you avoid scanning slow network mounts. This feature is not supported on all platforms.",
+    },
+
+    {
+      name: ["-p", "--path-to-ignore"],
+      description: "Provide a path to a specific .ignore file.",
+      args: {
+        name: "STRING",
+        description: "The path to the ignore file",
```
Comment (QuiiBz) [inline dev/ag.ts:301]: Looks like we can add a template of `filepaths`

File: dev/ag.ts [hunk 9]
```diff
@@ -0,0 +1,433 @@
...
+      },
+    },
+
+    {
+      name: "--pager",
+      description:
+        "Use a pager such as less. Use --nopager to override. This option is also ignored if output is piped to another program.",
+      args: {
+        name: "COMMAND",
+        description: "The pager",
```
Comment (QuiiBz) [inline dev/ag.ts:311]: Could we add [suggestions](https://fig.io/docs/reference/arg#suggestions) to this argument? (just to name a few, more, less, most...)

**Review by QuiiBz — CHANGES_REQUESTED** [review]
Thanks a lot for this spec!

> This is my first time writing autocomplete specs so let me know if there's anything I should've done differently!

That's actually really good! I left some comments to improve further the spec. Also, could you rebase/merge `master`, because we added a new ESLint rule (#523) for descriptions, and I think there are some issues with it in this spec.

**Comment by RSO** [comment]
> > Is there a way to get the autocomplete to show certain options above others?
> 
> Yep! You can use the [`priority`](https://fig.io/docs/reference/option#priority) property.

It seems to me that the priority only is available on the `option` and not on the `arg`? In my case, I'd like the args to be shown above the options if that makes sense!

**Comment by QuiiBz** [comment]
> It seems to me that the priority only is available on the `option` and not on the `arg`? In my case, I'd like the args to be shown above the options if that makes sense!

You're right, `priority` is currently only available for `options`. We might need to add it to `args`.

**Review by QuiiBz — APPROVED** [review]
LGTM, thanks a lot for this amazing spec!

**Pull request merged**